sims/verilator/Makefile

```diff
@@ -1,112 +1,175 @@
+                              --prof-threads --prof-cfuncs,\
+                              $(if $(filter $(VL8_PROFILE),threads),\
+                                --prof-threads,))
+
+VL8_THREADS ?= 1
+RUNTIME_THREADS := --threads $(VL8_THREADS) --threads-dpi all
+
+VL8_FST_MODE ?= 0
+TRACING_OPTS := $(if $(filter $(VL8_FST_MODE),0),--trace,--trace-fst-thread)
+TRACING_CFLAGS := $(if $(filter $(VL8_FST_MODE),0),,-DCY_FST_TRACE)
+
+#----------------------------------------------------------------------------
+# verilation configuration/optimization
+#----------------------------------------------------------------------------
+VMODE := -O3 --x-assign fast --x-initial fast --noassert \
+          --output-split 10000 --output-split-cfuncs 10000
 
 # default flags added for ariane
 ARIANE_VERILATOR_FLAGS = \
-	--unroll-count 256 \
-	-Werror-PINMISSING \
-	-Werror-IMPLICIT \
-	-Wno-PINCONNECTEMPTY \
-	-Wno-ASSIGNDLY \
-	-Wno-DECLFILENAME \
-	-Wno-UNUSED \
-	-Wno-UNOPTFLAT \
-	-Wno-BLKANDNBLK \
-	-Wno-style \
-	-Wall
+  --unroll-count 256 \
+  -Werror-PINMISSING \
+  -Werror-IMPLICIT \
+  -Wno-PINCONNECTEMPTY \
+  -Wno-ASSIGNDLY \
+  -Wno-DECLFILENAME \
+  -Wno-UNUSED \
+  -Wno-UNOPTFLAT \
+  -Wno-BLKANDNBLK \
+  -Wno-style \
+  -Wall
 
 # normal flags used for chipyard builds (that are incompatible with ariane)
-CHIPYARD_VERILATOR_FLAGS = \
-	--assert
+#CHIPYARD_VERILATOR_FLAGS := --assert
+CHIPYARD_VERILATOR_FLAGS := 
+
+# options dependent on whether ariane or chipyard is used
+PLATFORM_OPTS := $(shell if ! grep -iq "module.*ariane" $(build_dir)/*.*v; \
+                   then echo "$(CHIPYARD_VERILATOR_FLAGS)"; \
+                   else echo "$(ARIANE_VERILATOR_FLAGS)"; fi)
 
 # Use --timescale to approximate timescale behavior of pre-4.034
-TIMESCALE_OPTS := $(shell verilator --version | perl -lne 'if (/(\d.\d+)/ && $$1 >= 4.034) { print "--timescale 1ns/1ps"; }')
-VERILATOR_NONCC_OPTS = \
-	$(TIMESCALE_OPTS) \
-	--top-module $(VLOG_MODEL) \
-	-Wno-fatal \
-	$(shell if ! grep -iq "module.*ariane" $(build_dir)/*.*v; then echo "$(CHIPYARD_VERILATOR_FLAGS)"; else echo "$(ARIANE_VERILATOR_FLAGS)"; fi) \
-	--output-split 10000 \
-	--output-split-cfuncs 100 \
-	--max-num-width 1048576 \
-	-f $(sim_common_files) \
-	$(sim_vsrcs)
-
-VERILATOR_DEFINES = \
-	+define+PRINTF_COND=\$$c\(\"verbose\",\"\&\&\"\,\"done_reset\"\) \
-	+define+STOP_COND=\$$c\(\"done_reset\"\)
-
-VERILATOR_OPTS = $(VERILATOR_CC_OPTS) $(VERILATOR_NONCC_OPTS) $(VERILATOR_DEFINES) $(EXTRA_SIM_SOURCES)
-
-#########################################################################################
+TIMESCALE_OPTS := $(shell verilator --version | perl -lne \
+  'if (/(\d.\d+)/ && $$1 >= 4.034) { print "--timescale 1ns/1ps"; }')
+
+VERILATOR_NONCC_OPTS := \
+  $(RUNTIME_PROFILING_VFLAGS) \
+  $(RUNTIME_THREADS) \
+  $(VMODE) \
+  $(PLATFORM_OPTS) \
+  $(TIMESCALE_OPTS) \
+  --top-module $(VLOG_MODEL) \
+  +define+PRINTF_COND=\$$c\(\"verbose\",\"\&\&\"\,\"done_reset\"\) \
```

Comment:
I think it was nice to have the `defines` separated out into their own variable.

Reply:
I think we should move to this like it was done for the ongoing hammer sim PR.
Hopefully we can DRY some of the different simulator args that way.

Reply:
done